tests/serac_component_bc.cpp

```diff
@@ -135,13 +132,9 @@ TEST(component_bc, qs_attribute_solve)
   // Define the solver object
   NonlinearSolidSolver solid_solver(2, pmesh);
 
-  // define a boundary attribute array and initialize to 0
-  std::vector<int> ess_x_bdr(pmesh->bdr_attributes.Max(), 0);
-  std::vector<int> ess_y_bdr(pmesh->bdr_attributes.Max(), 0);
-
   // boundary attribute 1 (index 0) is fixed (Dirichlet) in the x direction
-  ess_x_bdr[0] = 1;
-  ess_y_bdr[1] = 1;
+  std::set<int> ess_x_bdr = {1};
```

Comment:
School me please, what does 1 vs 2 represent/mean?

Reply:
The input mesh has different markers (1,2,etc) for each part of the boundary. This is saying the boundary with the '1' markers are of the essential x boundary condition type. You could have multiple markers for a single boundary, so I thought a set was more appropriate. The mfem native way of dealing with this is a vector of length maximum number of attributes that is zero if an attribute isn't active and 1 if it is. I thought this was much more user friendly.